src/vs/workbench/contrib/scm/browser/repositoryPane.ts

```diff
@@ -77,6 +77,34 @@ import { ILabelService } from 'vs/platform/label/common/label';
 
 type TreeElement = ISCMResourceGroup | IResourceNode<ISCMResource, ISCMResourceGroup> | ISCMResource;
 
+function splitMatches(uri: URI, filterData: FuzzyScore | undefined): [IMatch[] | undefined, IMatch[] | undefined] {
+	let matches: IMatch[] | undefined;
+	let descriptionMatches: IMatch[] | undefined;
+
+	if (filterData) {
+		matches = [];
+		descriptionMatches = [];
+
+		const fileName = basename(uri);
+		const allMatches = createMatches(filterData);
+
+		for (const match of allMatches) {
+			if (match.start <= fileName.length && match.end <= fileName.length) {
```

Comment:
Might just be enough to do this:

```suggestion
			if (match.end <= fileName.length) {
```

Since `start < end`.